packages/mobile/src/identity/reducer.ts

```diff
@@ -149,39 +155,38 @@ export const reducer = (
     case Actions.IMPORT_CONTACTS:
       return {
         ...state,
-        isLoadingImportContacts: true,
         askedContactsPermission: true,
-        contactMappingProgress: { current: 0, total: 0 },
+        importContactsProgress: { status: ImportContactsStatus.Prepping, current: 0, total: 0 },
       }
-    case Actions.UPDATE_IMPORT_SYNC_PROGRESS:
+    case Actions.UPDATE_IMPORT_CONTACT_PROGRESS:
+      const curProgress = state.importContactsProgress
       return {
         ...state,
-        contactMappingProgress: { current: action.current, total: action.total },
-      }
-    case Actions.INCREMENT_IMPORT_SYNC_PROGRESS:
-      return {
-        ...state,
-        contactMappingProgress: {
-          current: state.contactMappingProgress.current + action.increment,
-          total: state.contactMappingProgress.total,
+        importContactsProgress: {
+          current: action.current ?? curProgress.current,
+          total: action.total ?? curProgress.total,
+          status: action.status ?? curProgress.status,
         },
       }
     case Actions.END_IMPORT_CONTACTS:
+      const { success } = action
       return {
         ...state,
-        isLoadingImportContacts: false,
-        contactMappingProgress: action.success
-          ? {
-              current: state.contactMappingProgress.total,
-              total: state.contactMappingProgress.total,
-            }
-          : state.contactMappingProgress,
+        importContactsProgress: {
+          ...state.importContactsProgress,
+          status: success ? ImportContactsStatus.Done : ImportContactsStatus.Failed,
+        },
       }
     case Actions.DENY_IMPORT_CONTACTS:
       return {
         ...state,
         askedContactsPermission: true,
       }
+    case Actions.ADD_CONTACT_MATCHES:
+      return {
+        ...state,
+        matchedContacts: [...state.matchedContacts, ...action.matches],
```

Comment:
If we search for matches across the full contact list every time, should new matches overwrite previous matches instead of adding to them? Thinking about a scenario where a user deletes contacts from their address book yet they are still showing up as a matched contact here.

Reply:
Yeah good question. We currently only allow matchmaking to be done once, so it's essentially equivalent if we went with a `SET` vs an `ADD` here. 
We want to eventually allow users to do a subsequent matchmaking session on just their new contacts, in which case an `ADD` approach will be better. So I'm inclined to leave this as it for now. WDYT?

Reply:
just wondering if using an object and indexing by phone number of contact id might be useful in terms of deduplicating and also quick lookups?

Reply:
The num of matches will likely be very small but okay will do